modules/navier_stokes/include/auxkernels/RANSYPlusAux.h

```diff
@@ -0,0 +1,63 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AuxKernel.h"
+#include "INSFVVelocityVariable.h"
+/**
+ * Computes wall y+ based on wall functions.
+ */
+class RANSYPlusAux : public AuxKernel
+{
+public:
+  static InputParameters validParams();
+
+  virtual void initialSetup() override;
+
+  RANSYPlusAux(const InputParameters & parameters);
+
+protected:
+  virtual Real computeValue() override;
+
+  /// the dimension of the simulation
+  const unsigned int _dim;
+
+  /// x-velocity
+  const Moose::Functor<ADReal> & _u_var;
+  /// y-velocity
+  const Moose::Functor<ADReal> * _v_var;
+  /// z-velocity
+  const Moose::Functor<ADReal> * _w_var;
+
+  /// Turbulent kinetic energy
+  const Moose::Functor<ADReal> & _k;
```

Comment:
Technically `k` is only needed when `_wall_treatment == "neq"`. You could consider making this a pointer and then reading this parameter only when `neq` is selected for the wall treatment

Reply:
Capability added.